packages/crawler/src/crawler/crawler-engine.ts

```diff
@@ -16,38 +16,40 @@ export type ApifyMainFunc = (userFunc: Apify.UserFunc) => void;
 
 export class CrawlerEngine {
     public constructor(
+        private readonly logger: Logger,
         private readonly pageProcessorFactory: PageProcessorFactory = new PageProcessorFactory(),
         private readonly crawlerFactory: CrawlerFactory = new CrawlerFactory(),
         private readonly resourceCreator: ResourceCreator = new ApifyResourceCreator(),
         private readonly runApify: ApifyMainFunc = Apify.main,
     ) {}
 
     public async start(crawlerRunOptions: CrawlerRunOptions): Promise<void> {
-        if (!isNil(crawlerRunOptions.localOutputDir)) {
-            this.setLocalOutputDir(crawlerRunOptions.localOutputDir);
-        }
+        this.setEnvVars(crawlerRunOptions.localOutputDir);
 
         const requestList = await this.resourceCreator.createRequestList(crawlerRunOptions.existingUrls);
         const requestQueue = await this.resourceCreator.createRequestQueue(crawlerRunOptions.baseUrl);
 
-        const pageProcessor = this.pageProcessorFactory.createPageProcessor({
-            requestQueue,
-            crawlerRunOptions,
-        });
+        const pageProcessor = this.pageProcessorFactory.createPageProcessor(
+            {
+                requestQueue,
+                crawlerRunOptions,
+            },
+            this.logger,
```

Comment:
can you make this a Singleton class and pass it in the parameter of each class? I'm planning to switch to inverse dependence injection later if we got time at the end of the feature.